lib/puma/server.rb

```diff
@@ -528,7 +528,8 @@ def lowlevel_error(e, env, status=500)
       end
 
       if @leak_stack_on_error
-        [status, {}, ["Puma caught this error: #{e.message} (#{e.class})\n#{e.backtrace.join("\n")}"]]
+        backtrace = e.backtrace.nil? ? '<no backtrace available>' : e.backtrace.join("\n")
```

Comment:
Style nitpick: in this case, the ternary doesn't make a lot of sense. Say `e.backtrace` is `false` (i mean, it can't be, but say that it is): then we're going to call `false.join("\n")`. What you really care about is the ability to `respond_to?` the `join` method, so you should check for that instead.

Reply:
Current Ruby docs for [`Exception#backtrace`](https://msp-greg.github.io/ruby_trunk/Core/Exception.html#backtrace-instance_method) state:

'In the case no backtrace has been set, nil is returned'

That seems imply that the method always exists?

Reply:
that's why I said it's a style nitpick and not a correctness nitpick 😄 IMO anything passing the test in the ternary should work

Reply:
I guess we could check whether the backtrace returns an array?

Sometimes with stuff like this it may better to keep the scope to what Ruby docs state, so if the Ruby API changes, we **might** see it in tests?  Believe it or not, I've seen things like that happen working with Ruby master and even OpenSSL.  IOW, keep the test very narrow, and edges cases appear.

JFYI, I have no style...